### PR TITLE
[TF-648] Reformat string to comply and fix JSON-LD error

### PR DESCRIPTION
**Pull request by zourdyzou**

### CODE REVIEW

**Comment by zourdyzou** [comment]
https://user-images.githubusercontent.com/69587933/162691480-f5d7a7fd-1ada-464e-825d-a862151143c1.mov

@oleg-semyonov 

I think we don't need to check for the single quotes as the JSON-LD string itself uses double-quoted string as a wrapper

File: components/StorySeo/StorySeo.tsx [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
+    title: string;
+    description: string;
+};
+
 function StorySeo({ story }: Props) {
     const { title, subtitle, published_at, updated_at, author, oembed, newsroom, summary } = story;
 
     const authorName = author?.display_name || author?.email || 'Unknown';
     const description = subtitle || summary;
+    const modifiedString = escapedQuotedString({ title, description }) as ManipulatedString;
```
Comment (oleg-semyonov) [inline components/StorySeo/StorySeo.tsx:20]: This is an overkill. It's much easier to make a function that manipulates a string, but also your object would be recreated on each render, resulting in unwanted re-renders.

**Comment by zourdyzou** [comment]
https://user-images.githubusercontent.com/69587933/162694032-5aa6f783-ab87-4dca-b30e-1c1851dcd018.mov

tested the JSON-LD schema

**Review by oleg-semyonov — CHANGES_REQUESTED** [review]
Can't approve this one, see my comments.

File: components/StorySeo/StorySeo.tsx [hunk 2]
```diff
@@ -50,14 +58,14 @@ function StorySeo({ story }: Props) {
             />
             <ArticleJsonLd
                 url={oembed.url}
-                title={title}
+                title={modifiedString.title}
```
[the remark this answers is not in the document]
Comment (oleg-semyonov) [reply components/StorySeo/StorySeo.tsx:61]: I also see that you're testing the schema with already escaped string. I'm curious to see what happens with the original string. Also, did the story in the original ticket also contain different opening and closing quote marks?

[the remark this answers is not in the document]
Comment (zourdyzou) [reply components/StorySeo/StorySeo.tsx:61]: yep I've checked this one -> https://github.com/garmeeh/next-seo/issues/215

File: components/StorySeo/StorySeo.tsx (hunk 1 above)
Comment (zourdyzou) [reply components/StorySeo/StorySeo.tsx:20]: > your object would be recreated on each render, resulting in unwanted re-renders.

That is exactly what I thought before and we could directly change it in the `ArticleJsonLd` with RegEx but I'm not sure if that is a feasible solution or not

Comment (oleg-semyonov) [reply components/StorySeo/StorySeo.tsx:20]: The simpler solution is always feasible

File: components/StorySeo/StorySeo.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (oleg-semyonov) [reply components/StorySeo/StorySeo.tsx:61]: I spent 3 minutes browsing through the link you've sent, and I found:
- A simple and robust fix for escaping strings: https://github.com/garmeeh/next-seo/issues/711#issuecomment-888330041
- There is a major v5 version which should fix the issue: https://github.com/garmeeh/next-seo/issues/711#issuecomment-1024868596

The latter is what I asked you to check first before writing any new implementations.

File: components/StorySeo/StorySeo.tsx (hunk 1 above)
Comment (zourdyzou) [reply components/StorySeo/StorySeo.tsx:20]:
```js
title={JSON.stringify(title).slice(1, -1)}
```
what do you think about the approach above? this also works

Comment (oleg-semyonov) [reply components/StorySeo/StorySeo.tsx:20]: This is what I've linked to you in the comment below

File: components/StorySeo/StorySeo.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (zourdyzou) [reply components/StorySeo/StorySeo.tsx:61]: > did the story in the original ticket also contain different opening and closing quote marks?

I'm not sure what you mean by different opening and closing quote marks but yes there is a different quote like this:
“Sekswerk is geen abnormaal beroep" but JSON-LD did not see that as a problem, see the video below ⬇️ 

https://user-images.githubusercontent.com/69587933/162701034-644e9fe1-6bbc-42bd-868c-3d532e735ef8.mov


https://prezly.slack.com/archives/C029LTE0M/p1649060623734269?thread_ts=1649060313.720369&cid=C029LTE0M

> I also see that you're testing the schema with an already escaped string. I'm curious to see what happens with the original string

https://user-images.githubusercontent.com/69587933/162699877-af3e6791-8f1b-4a6a-a0b0-d0229cdc84a2.mov

File: components/StorySeo/StorySeo.tsx (hunk 1 above)
Comment (zourdyzou) [reply components/StorySeo/StorySeo.tsx:20]: fixed 👍🏼

Comment (zourdyzou) [reply components/StorySeo/StorySeo.tsx]: Screenshot 2022-04-11 at 11 37 40

**Comment by zourdyzou** [comment]
Screenshot 2022-04-11 at 11 37 40

@oleg-semyonov

**Review by oleg-semyonov — APPROVED** [review]
Looks about right. Please compare with the current SEO output (meta tags, etc.) on production Bea newsroom, to confirm that nothing else was changed. After that, good to merge.

**Comment by zourdyzou** [comment]
checking in now cap @oleg-semyonov

**Pull request merged**